app/services/rent_service.rb

```diff
@@ -0,0 +1,44 @@
+module RentService
+  def self.reservation(user, rent)
+    Rent.transaction do
```

Comment:
Here transaction is needed, because you want to group rent#save together with rent#overbook.